Rakefile

```diff
@@ -33,3 +49,35 @@ task :validate => [:build] do
   # --url-swap          # Adjust for Jekyll's baseurl. See https://github.com/gjtorikian/html-proofer/issues/618 for more.
   sh('bundle exec htmlproofer _site --assume-extension --disable-external --empty-alt-ignore --allow-hash-href --url-swap "^/docs/:/"')
 end
+
+task :validate_sidebar_tree do # => [:build] do
+  # There are lots of things which this could validate, however we assume that
+  # most changes will be eyeballed by a human. We therefore just check the most
+  # nuanced case -- that the url must be an exact match for its target page.
+
+  def check_url(url)
+    if url.end_with? "/" then
+      raise "Imprecise target url '#{url}' in sidebar (did you mean '#{url[..-2]}'?)\n\n" unless File.directory?("_site#{url}")
+    else
+      raise "Imprecise target url '#{url}' in sidebar (did you mean '#{url}/'?)\n\n" unless File.file?("_site#{url}.html")
+    end
```

Comment:
I'm not sure this is going to work as we want? If I'm reading this right, urls which end with `/` and where a directory is present will fall through to the `elsif` check for a `.html` file in that directory. Since such a file will almost certainly not be present they would then error (incorrectly).

Reply:
Just tested this and it does indeed error in the surprising way I've described.

Reply:
I'm a bit confused as all I did was move the postfix unless to its above `if` so surely what you wrote beforehand should also break in the same way(?)

Reply:
Note that the postfix `unless` is (deliberately) decoupled from the `if`/`else` structure in the original. Your changed proposal couples them, which introduces the issue described.

Reply:
Ok, I understand what you're saying now.